packages/material-ui/src/Link/Link.js

```diff
@@ -44,35 +46,68 @@ export const styles = {
     '&::-moz-focus-inner': {
       borderStyle: 'none', // Remove Firefox dotted outline.
     },
+    '&$focusVisible': {
+      outline: 'auto',
+    },
   },
+  /* Styles applied to the root element if the link is keyboard focused. */
+  focusVisible: {},
 };
 
 const Link = React.forwardRef(function Link(props, ref) {
   const {
     classes,
     className,
-    component = 'a',
     color = 'primary',
+    component = 'a',
+    onBlur,
+    onFocus,
     TypographyClasses,
     underline = 'hover',
     variant = 'inherit',
     ...other
   } = props;
 
+  const linkRef = React.useRef();
+  // TODO: Olivier originally included an arg of `() => linkRef.current.ownerDocument`. Am i missing something?
+  const { isFocusVisible, onBlurVisible } = useIsFocusVisible();
```

Comment:
Could you rebase it to use the latest logic? #15851

Reply:
`useFocusVisible` is now returning a ref that should be attached to the host instance with the polyfill. When I re-evaluate the API I will properly annotate it so that you don't have to look into the implementation or other callsites.

Reply:
Thanks for the heads up. Should be good now I believe.